components/NewsletterSubscribe.js

```diff
@@ -1,24 +1,95 @@
+import { useState } from "react";
 import Button from "./buttons/Button";
 import Heading from "./typography/Heading";
 import Paragraph from "./typography/Paragraph";
+import Loader from "./Loader";
 import { useTranslation } from "../lib/i18n";
 
-export default function NewsletterSubscribe ({
+export default function NewsletterSubscribe({
   className = 'p-8 text-center',
   dark = false,
-  title = 'Subscribe to our newsletter to receive news about AsyncAPI.',
-  subtitle = 'We respect your inbox. No spam, promise ✌️',
   type,
 }) {
 
-  const { t } = useTranslation('common'); 
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [status, setStatus] = useState("normal");
+
+  const { t } = useTranslation('common');
 
   const headTextColor = dark ? 'text-white' : ''
   const paragraphTextColor = dark ? 'text-gray-300' : ''
 
+  const handleSubmit = (e) => {
+    setStatus("loading");
+    e.preventDefault()
+    const data = {
+      name: name,
+      email: email
+    }
+
+    fetch("/.netlify/functions/newsletter_subscription", {
+      method: "POST",
+      body: JSON.stringify(data),
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    }).then((res) => {
+      if (res.status === 200) {
+        setFormStatus("success");
+      } else {
+        setFormStatus("error");
+      }
+      return res.json()
+    }).then((data) => console.log(data))
```

Comment:
This console.log needs to be removed? Not an issue but redundant.

Reply:
I added this because it's important to note that it is a dual promise. We will get proper data after resolving both promises only. That's why I added it. Maybe later on, new functionalities will be added upon this.

Reply:
Since this is not an async function, this makes sense